dali/test/python/test_operator_random_bbox_crop.py

```diff
@@ -0,0 +1,238 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.ops as ops
+import nvidia.dali.types as types
+import nvidia.dali as dali
+from nvidia.dali.backend_impl import TensorListGPU
+import numpy as np
+import os
+
+class RandomBBoxCropSynthDataPipeline(Pipeline):
+    def __init__(self, device, batch_size,
+                 boxes,
+                 labels = None,
+                 thresholds = [0, 0.01, 0.05, 0.1, 0.15],
+                 scaling = [0.3, 1.0],
+                 aspect_ratio = [0.5, 2.0],
+                 ltrb = True,
+                 num_attempts = 100,
+                 allow_no_crop = False,
+                 input_shape = None,
+                 crop_shape = None,
+                 num_threads=1, device_id=0, num_gpus=1):
+        super(RandomBBoxCropSynthDataPipeline, self).__init__(
+            batch_size, num_threads, device_id, seed=1234)
+        self.device = device
+        self.boxes = boxes
+        self.labels = labels
+        self.boxes_inputs = ops.ExternalSource()
```

Comment:
Why not use a new ExternalSource API?

Reply:
Because this was written before the new API came :) I can try to adapt it